packages/angular/src/utils/mfe/with-module-federation.ts

```diff
@@ -0,0 +1,137 @@
+import {
+  SharedLibraryConfig,
+  sharePackages,
+  shareWorkspaceLibraries,
+} from './mfe-webpack';
+import { createProjectGraphAsync } from '@nrwl/workspace/src/core/project-graph';
+import { readWorkspaceJson } from '@nrwl/workspace';
+import { joinPathFragments } from '@nrwl/devkit';
+import ModuleFederationPlugin = require('webpack/lib/container/ModuleFederationPlugin');
+
+export type MFERemotes = string[] | [remoteName: string, remoteUrl: string][];
+
+export interface MFEConfig {
+  name: string;
+  remotes?: MFERemotes;
+  exposes?: Record<string, string>;
+  shared?: (
+    libraryName: string,
+    library: SharedLibraryConfig
+  ) => SharedLibraryConfig | false;
+}
+
+async function getDependentPackagesForProject(name: string) {
+  const projectGraph = await createProjectGraphAsync();
+  return projectGraph.dependencies[name].reduce(
+    (dependencies, dependency) => {
+      const workspaceLibraries = dependencies.workspaceLibraries;
+      const npmPackages = dependencies.npmPackages;
+
+      if (dependency.target.startsWith('npm')) {
+        npmPackages.push(dependency.target.replace('npm:', ''));
+      } else {
+        workspaceLibraries.push(dependency.target);
+      }
+
+      return {
+        workspaceLibraries,
+        npmPackages,
+      };
+    },
+    { workspaceLibraries: [], npmPackages: [] }
+  );
+}
+
+function determineRemoteUrl(remote: string) {
+  const workspace = readWorkspaceJson();
+  return joinPathFragments(
+    workspace.projects[remote].targets.serve.options.publicHost,
+    'remoteEntry.mjs'
+  );
+}
+
+function mapRemotes(remotes: MFERemotes) {
+  const mappedRemotes = {};
+
+  for (const remote of remotes) {
+    if (Array.isArray(remote)) {
+      mappedRemotes[remote[0]] = remote[1];
+    } else if (typeof remote === 'string') {
+      mappedRemotes[remote] = determineRemoteUrl(remote);
+    }
+  }
+
+  return mappedRemotes;
+}
+
+export async function withModuleFederation(options: MFEConfig) {
```

Comment:
Do we want to make the return value of type `(config: webpack.Configuration) => webpack.Configuration`? 

An example usage would be:
```
// apps/myapp/webpack.config.js
const withModuleFederation = require('@nrwl/angular/module-federation');

module.exports = withModuleFederation({
   // MFE options...
});
```

Then in the project config:

```json
{
  "targets": {
     "build": {
       ...
       "options": {
         ...
         "webpackConfig": "apps/myapp/webpack.config.js"
       }
     }
  }
}
```

We'll need this for React (and other React metaframeworks like Next.js, Remix) since there are existing loaders and plugins needed.